dot/parachain/candidate-validation/register.go

```diff
@@ -0,0 +1,12 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package candidatevalidation
+
+func Register(overseerChan chan<- any) (*CandidateValidation, error) {
+	candidateValidation := CandidateValidation{
+		SubsystemToOverseer: overseerChan,
+	}
+
+	return &candidateValidation, nil
+}
```

Comment:
This looks like a constructor, can it be renamed to `NewCandidateValidation` and looks like it does not return any error, then you can return just `*CandidateValidation`

Reply:
Updated.